scheduler/src/modules/transactions/transaction.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Transaction } from './transaction.entity';
+import { TransactionDetail } from './transaction-detail.entity';
+import { Cron } from '@nestjs/schedule';
+import { LogAsyncMethodExecution } from 'src/common/decorator/log-async-method-execution.decorator';
+import { generate } from 'src/helper/generator.helper';
+import { Like, Repository } from 'typeorm';
+import { BcHoliday } from './holiday.entity';
+
+@Injectable()
+export class TransactionService {
+  private holidays: string[] = [];
+  constructor(
+    @InjectRepository(Transaction)
+    private readonly transactionRepository: Repository<Transaction>,
+    @InjectRepository(TransactionDetail)
+    private readonly cfsTransactionDetailRepo: Repository<TransactionDetail>,
+    @InjectRepository(BcHoliday)
+    private readonly holidayRepository: Repository<BcHoliday>,
+  ) {
+    void this.initializeHolidays();
+  }
+
+  private async initializeHolidays(): Promise<void> {
+    this.holidays = await this.getBcHolidays();
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
+    return this.transactionRepository.find();
+  }
+
+  async getHolidays(): Promise<BcHoliday[]> {
+    const now: Date = new Date();
+    const year = now.getFullYear().toString(); // Convert year to string
+    
+    return this.holidayRepository.find({
+      where: { holidayDate: Like(`%${year}%`) }
+    });
+  }
+
+  async getTransactionDetails(): Promise<Transaction[]> {
+    const transactions = await this.transactionRepository.createQueryBuilder('transaction')
+    .innerJoinAndSelect('TransactionDetail', 'detail', 'transaction.TRANSACTION_ID = detail.TRANSACTION_ID')
+    .where('detail.CFS_FILE_STATUS_TYPE = :status', { status: 'READY' })
+    .getMany();
+
+    if (transactions.length > 0) {
+      console.log(transactions);
+      return transactions;
+    } else {
+      console.log('No transactions found with status READY');
+      return [];
+    }
+  }
+
+  async updateCfsFileStatusType(fileName: string): Promise<void> {
+    const currentDate = new Date();
+    const currentUTCTimestamp = currentDate.toISOString();
+    const transactionDetails: TransactionDetail[] = await this.cfsTransactionDetailRepo.find({ where: { CFS_FILE_STATUS_TYPE: 'READY' } });
```

Comment:
Implement error handling for file uploads. Update the file status to 'SENT' only upon successful file upload completion. Instead of checking for 'transaction_status = READY', we should identify transactions by their corresponding transaction IDs that are being sent. Relying solely on 'transaction_status = READY' might lead to updating transactions that were added after polling the batch intended for processing.